### PR TITLE
- Added `.keep_all` argument to `duplicates()`

### PR DESCRIPTION
**Pull request by joethorley**
the distinct function is looking really good.

this is a PR with some more testing, a tiny bit of refactoring and the addition of the `.keep_all` argument.

We need to figure out how we want to deal with missing values. Probably worth seeing what distinct does with them.

```
data <- tibble::tibble(x = c(1,2,NA,1), y = c(1,1,NA,NA))
duplicates(data)
```

### CODE REVIEW

**Review by dunkenwg — APPROVED** [review]
`distinct()` treats NA as it does any other value. If two rows have NA in the same column but different values in other columns, they are considered distinct. If two rows have exactly the same values, including NA, they are not considered distinct.

`duplicates()` handles NA in a similar way. If two rows have NA in the same column but different values in other columns, they are not considered duplicates. If two rows have exactly the same values, including NA, they are considered duplicates.

I've also added two more tests for handling columns with missing values.

**Comment by joethorley** [comment]
Great - I agree with how missing values are being handled! Good work.

I've added a test for data sets with no duplicates.

**Comment by joethorley** [comment]
@aylapear - once you approve then @dunkenwg can merge.

File: R/duplicates.R [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
-#' @param ... Optional column names to use when searching for duplicate rows in specific columns.
-#' @return A tibble containing only duplicate rows.
+#' @param ... Optional variables to use when determining non-uniqueness.
+#'  If omitted, will use all variables in the data frame.
+#' @param .keep_all A flag specifying whether to keep all variables in .data.
+#' @return The original data frame with only non-unique rows.
 #' @export
-duplicates <- function(.data, ...) {
+#' @examples 
+# add_column ---------------------------------
```
Comment (aylapear) [inline R/duplicates.R]: What is this line of code for?

**Review by aylapear — COMMENTED** [review]
Everything looks good except the one line of code I provided comment on.

File: R/duplicates.R (hunk 1 above)
Comment (joethorley) [reply R/duplicates.R]: Good catch - that `# add_column ------------` line is a paste error!

Comment (joethorley) [reply R/duplicates.R]: I've just pushed a fix without it

**Review by aylapear — APPROVED** [review]

**Pull request merged**